src/main/java/org/jetbrains/research/intellijdeodorant/ide/refactoring/PolymorphismRefactoring.java

```diff
@@ -0,0 +1,607 @@
+        }
+    }
+
+    protected PsiExpression constructExpression(DefaultMutableTreeNode node) {
+        Object object = node.getUserObject();
+        if (object instanceof PsiJavaToken) {
+            PsiJavaToken operator = (PsiJavaToken) object;
+            DefaultMutableTreeNode leftChild = (DefaultMutableTreeNode) node.getChildAt(0);
+            DefaultMutableTreeNode rightChild = (DefaultMutableTreeNode) node.getChildAt(1);
+            PsiExpression leftExpression = constructExpression(leftChild);
+            PsiExpression rightExpression = constructExpression(rightChild);
+            PsiBinaryExpression binaryExpression = (PsiBinaryExpression)
+                    elementFactory.createExpressionFromText("a" + operator.getText() + "b", null);
+            binaryExpression.getLOperand().replace(leftExpression);
+            binaryExpression.getROperand().replace(rightExpression);
+            return binaryExpression;
+        } else if (object instanceof PsiExpression) {
+            return (PsiExpression) object;
+        }
+        return null;
+    }
+
+    protected boolean sourceTypeRequiredForExtraction() {
+        return Stream.of(
+                typeCheckElimination.getAccessedFields(),
+                typeCheckElimination.getAssignedFields(),
+                typeCheckElimination.getAccessedMethods(),
+                typeCheckElimination.getAccessedMethods(),
+                typeCheckElimination.getSuperAccessedFieldBindings(),
+                typeCheckElimination.getSuperAssignedFieldBindings()
+        ).flatMap(Collection::stream).anyMatch(element ->
+                element.getModifierList() == null || !element.getModifierList().hasModifierProperty(PsiModifier.STATIC)
+        );
+    }
+
+    protected void addImports(PsiImportList importList, Set<PsiType> requiredImportDeclarations) {
+        for (PsiType typeBinding : requiredImportDeclarations) {
+            PsiClass resolvedClass = PsiUtil.resolveClassInType(typeBinding);
+            if (resolvedClass != null && resolvedClass.getContainingClass() == null && !PsiUtil.getPackageName(resolvedClass).isEmpty()) {
+                importList.add(elementFactory.createImportStatement(resolvedClass));
+            }
+        }
+    }
+
+    protected static PsiImportList getPsiImportList(PsiFile classFile) {
+        PsiElement[] children = classFile.getChildren();
+        for (PsiElement child : children) {
+            if (child instanceof PsiImportList) {
+                return (PsiImportList) child;
+            }
+        }
+        // will not happen
+        return null;
+    }
+
+    public Project getProject() {
+        return project;
+    }
+
+    public PsiClass getSourceTypeDeclaration() {
+        return sourceTypeDeclaration;
+    }
+
+    public PsiFile getSourceFile() {
+        return sourceFile;
+    }
+
+//	protected Expression generateDefaultValue(ASTRewrite sourceRewriter, AST ast, ITypeBinding returnTypeBinding) {
```

Comment:
?

Reply:
It was supposed to be useful for the `initializeReturnedVariableDeclaration` method which was commented.  I uncommented `initializeReturnedVariableDeclaration` (see my comment below) but decided to use `PsiTypesUtil.getDefaultValueOfType` instead of this code (which is now removed).